spec/dynamic/layout.md

```diff
@@ -0,0 +1,333 @@
+* `Rust`
+* `C`
+* `transparent`
+* `align(N)`
+* `packed(N)`
+
+[§](r[dynamic.layout.aggregate.repr-rust]) The field offsets, the size, and the alignment requirement of a `struct` or `union` definition with the `repr(Rust)` repr-attribute may be any valid value. `repr(Rust)` may not be combined with `repr(C)`
+
+[!NOTE]: This is the default layout behaviour of a `struct` or `union` definition.
+
+[!WARN]: The default layout for a `union` does not guarantee that each field starts at offset `0`.
+
+[§](r[dynamic.layout.aggregate.repr-c]) The field offsets of a `struct` or `union` definition with the `repr(C)` repr-attribute are the minimum valid offset for the field such that the field offsets in declaration order are increasing. The size and alignment requirement of such a `struct` or `union` definition are the minimum possible valid size with the given field offsets.
+
+[!NOTE]: For `union`s, the offset of each field is `0` and the size is the size of the largest field. For `struct`s, the offset of the first field is `0`, and each subsequent field's offset is the offset of the previous field, plus it's size, plus the minimum number such that the offset is aligned to the field's alignment requirement.
+
+[§](r[dynamic.layout.aggregate.transparent]) The `repr(transparent)` repr-attribute may be applied to a `struct` definition with no other repr-attribute and with at most 1 field that has a size greater than `0` or an alignment requirement greater than `1`. If such a field exists, the `struct` definition has an underlying type of the type of that field and that field has offset `0`. Otherwise, the `struct` definition has an underlying type of `()`. `repr(transparent)` may not be applied to a `union` definition.
+
+[!NOTE]: The offsets of 1-ZST fields of such a `struct` are not specified, but do not exceed the size of the underlying type.
+
+[§](r[dynamic.layout.aggregate.repr-align]) A `struct` or `union` definition with a `repr(align(N))` repr-attribute, where `N` is an integer power of 2 up to a target-defined value, has an alignment requirement of at least `N`. The `repr(align(N))` repr-attribute may not be combined with any `repr(packed(N))` repr-attributes.
+
+[§](r[dynamic.layout.aggregate.repr-packed]) A `struct` or `union` definition with a `repr(packed(N))` repr-attribute, where `N` is an integer power of 2 up to a target-defined value, sets the alignment requirement of each field to the smallest of the alignment requirement of the type and `N`. The `repr(packed(N))` repr-attribute may not be combined with any `repr(align(N))` repr-attributes.
+
+[!NOTE]: The repr-attribute `repr(packed(1))` may be written as `repr(packed)`
```

Comment:
I think this needs to be normative. `dynamic.layout.aggregate.repr-packed-short-form`